lyricsheets/models/karaoke.py

```diff
@@ -73,6 +141,30 @@ def calculate_char_kara_times(self, style: Style):
 
             sylAccLength += syllableCharLength
 
+    def calculate_sizing(self, style: Style):
+        fontScaler = FontScaler(style)
+
+        text = self.text
+        stripped = text.strip()
+
+        numPreSpaces = len(text) - len(text.lstrip())
+        numPostSpaces = len(text) - len(text.rstrip())
+
+        self.width = fontScaler.get_length(stripped)
+        self.preSpaceWidth = fontScaler.get_length(numPreSpaces * " ")
+        self.postSpaceWidth = fontScaler.get_length(numPostSpaces * " ")
```

Comment:
Technically this isn't entirely accurate if the stripped whitespace character are unicode spaces, for example.

Reply:
Will there ever be a situation where a non-basic space would be used in the lyrics somewhere (in Romaji or EN)?

Reply:
Probably not, but you might as well do `fontScaler.get_length(text[:numPreSpaces])` instead

Reply:
Updated.